pkg/redfish/rfcomponents.go

```diff
@@ -1830,6 +1846,51 @@ func (s *EpSystem) discoverComponentEPEthInterfaces() {
 			}
 		}
 	}
+
+	if foxconnMACWorkarround {
+		var mgr *EpManager = nil
+		var ok bool = false
+		// Get the first manager linked to in the system object
+		for _, oid := range s.ManagedBy {
+			mgr, ok = s.epRF.Managers.OIDs[oid.Basename()]
+			if ok {
+				break
+			}
+		}
+		// If no link to ManagedBy Manager in system object, just pick
+		// the first manager (there is likely only one)
+		if !ok {
+			for _, m := range s.epRF.Managers.OIDs {
+				mgr = m
+				break
+			}
+		}
+		if mgr != nil {
+			foundMac := false
+			for _, eth := range mgr.ENetInterfaces.OIDs {
+				if eth.EtherIfaceRF.Id == "eth0" {
+					bmcMac := eth.EtherIfaceRF.MACAddress
+					mac, err := GetOffsetMACString(bmcMac, int64(ParadiseMacWarInc))
+					if err != nil {
+						errlog.Printf("%s: failed to increment the bmc mac %s by %d, error: %s",
+							s.ID, bmcMac, ParadiseMacWarInc, err)
+						continue
+					}
+					mac = strings.ToLower(mac)
+					ethIDAddr := new(EthernetNICInfo)
+					ethIDAddr.MACAddress = mac
+					ethIDAddr.Description = "MAC computed from the BMC MAC via workaround"
+					s.EthNICInfo = append(s.EthNICInfo, ethIDAddr)
```

Comment:
Do we need a 's.MACAddr = ethIDAddr.MACAddress' here as well?

Reply:
What is the s.MACAddr?
Other places seem to call it the management mac

Reply:
Yeah, it's the primary MAC for the management network attached to the node.

Reply:
I wouldn't think that the server has a management network.
Management network means BMC to me.
Do you think it should be set?

Reply:
I think they mean the management network MAC.  If you look at the code above this in discoverComponentEPEthInterfaces(), it always sets s.MACAddr when it iterates through s.ENetInterfaces.OIDs.  It also refers to this in the comments above as the "main node MAC address".

Reply:
Most of the nodes on tyr don't have the MACAddr set.
I looked at a few other systems, and it seems that it is only set when there is a node that hits this workaround (the pervious workaround for other types of systems)
I guess I still think that it should not be set.
To see this run this command
```
cray hsm inventory componentEndpoints list --format json --type Node | jq -c '.ComponentEndpoints[] | { "ID":.ID, "MACAddr": .MACAddr }' | sort
```

Reply:
It was a good catch. It was my mistake to not make this workaround like the others. However, it seems like normal Nodes don't set that value, so after looking at it I think it shouldn't be set and the other workarounds are wrong.
This field may only be for NodeBMCs.

Reply:
I see what you mean.  Only the nodes I ran discover against with my modified code are showing it.  I'm not sure how I misunderstood that code block so poorly.  I will remove it and push up the change.  Thanks for the catch!